service/controller/aws/v14/resource/kubeconfig/resource.go

```diff
@@ -0,0 +1,104 @@
+package kubeconfig
+
+import (
+	"time"
+
+	"github.com/giantswarm/certs"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "kubeconfigv14"
+	// DefaultWatchTimeout is the time to wait on watches against the Kubernetes
+	// API before giving up and throwing an error.
+	DefaultWatchTimeout = 90 * time.Second
+)
+
+// Config represents the configuration used to create a new kubeconfig resource.
+type Config struct {
+	// Dependencies.
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+
+	// Settings.
+	CertsWatchTimeout time.Duration
+	ProjectName       string
+	ResourceNamespace string
+}
+
+// StateGetter implements the kubeconfig resource.
+type StateGetter struct {
+	// Dependencies.
+	certsSearcher certs.Interface
+	k8sClient     kubernetes.Interface
+	logger        micrologger.Logger
+
+	// Settings.
+	projectName       string
+	resourceNamespace string
+}
+
+// New creates a new configured index resource.
+func New(config Config) (*StateGetter, error) {
+	// Dependencies.
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	// Settings
+	if config.ProjectName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ProjectName not be empty", config)
+	}
+	if config.ResourceNamespace == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ResourceNamespace not be empty", config)
+	}
+	if config.CertsWatchTimeout == 0 {
```

Comment:
Why can't we use the already configured timeout of 5 secs in `service.go`?

90 seconds is way too long. We should use a 5 secs timeout and cancel the resource. We can retry on the next loop.